lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,39 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(hyped::io::I2c &i2c, hyped::core::ILogger &log) : log_(log), i2c_(i2c)
+{
+}
+
+std::optional<std::uint16_t> Temperature::read()
+{
+  if (i2c_.readByte(TEMPERATURE_1, STATUS) == 0) {
+    auto Temp_High = i2c_.readByte(TEMPERATURE_1, TEMP_H);
```

Comment:
readByte returns an optional, you'd want to check if the operation was successful

Reply:
Also const

Reply:
also variable names